lib/interactor/initializer.rb

```diff
@@ -9,28 +10,20 @@ def self.included(target_class)
     end
 
     module ClassMethods
-      def for(*args)
```

Comment:
I wonder if this makes IDEs like RubyMine perform worse or better with suggestions. @oleg-vinted maybe you know?

Reply:
The good thing is that this PR does a major version increase so that the projects wouldn't upgrade by accident. That in turn means that nobody will start using this version right away and the IDE suggestions won't be affected for most developers. If there's room for improvement in that area we can always discuss/change things in later PRs with minor version updates.

Reply:
Without these, RubyMine will have no idea interactors have `.for`, `.with`, and `.run` methods. If we're concerned with better IDE support, I think we'd be better off duplicating the code and removing as much metaprogramming as possible.

Reply:
Even though it was more interesting to do it this way I'm fine with duplication. In reviews there are always people who advocate for DRY code, but I guess there's a good reason to do it differently here. Admittedly I didn't really consider implications for the IDE.

Reply:
So... the issue with the non-dynamically generated methods and the changes to how method parameters are handled in ruby3 presents an interesting issue. I can't find any valid syntax that would allow retaining only one version of each method for both cases of regular params and keyword params. There is of course a solution, but it doesn't look pretty

```ruby
def for(*args)
  if args.size == 1 && args[0].class == Hash
    new(**args[0]).run
  else
    new(*args).run
  end
end
```

A single splat doesn't work for both cases, so additional checks are needed to determine if the thing that's being passed is actually a Hash and should be dealt with a double splat. The `...` parameter works only for regular params and doesn't correctly pass on the keyword params.

```ruby
def for(...)
  new(...).run
end
```

Personally, I think that the intermediate checks in the static variant are worse than the lack of syntax completion in IDEs. If there's another way I haven't found it yet, suggestions are welcome. What is your take on this matter? @tomasv @oleg-vinted

Reply:
I haven't tried anything myself, but I think if IDE compatibility is hard to keep, let's not worry about that. IDE support isn't that great anyway.